### PR TITLE
Update Fast DDS-Gen documentation [8708]

### PR DESCRIPTION
**Pull request by rsanchez15**

### CODE REVIEW

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/513/)

File: docs/fastddsgen/pubsub_app/includes/idl_file.rst [hunk 1]
```diff
@@ -1,12 +1,11 @@
...
-If you have followed the steps, you should have Fast DDS, Fast CDR, and Fast-RTPS-Gen installed.
+Creating the IDL file with the data type
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
 To build a minimal application, you must first define the Topic.
 To do this, the data type of the Topic is defined by means of an IDL file.
 An example of this is shown below. In this case, a single string is sufficient.
 Topics are explained in more detail in :ref:`dds_layer_topic`, while the topic data types to be defined using IDL are
-presented in :ref:`fastrtpsgen_intro`.
+presented in :ref:`dds_layer_definition_data_types`.
```
Comment (EduPonz) [inline docs/fastddsgen/pubsub_app/includes/idl_file.rst:8]: This paragraph is a little strange to read, a lot of very short sentences that may be combined in longer ones

File: docs/fastddsgen/pubsub_app/includes/first_app.rst [hunk 2]
```diff
@@ -1,21 +1,11 @@
...
-    .. code:: bash
-
-        source <path-to-Fast-DDS-workspace>/install/setup.bash
-
-    - However, if you have followed the **manual installation** the libraries are already accessible.
+Generate the Fast DDS source code
+***********************************
 
 The application files are generated using the following command.
 The `-example` option creates an example application, and the CMake files needed to build it.
```
Comment (EduPonz) [inline docs/fastddsgen/pubsub_app/includes/first_app.rst:11]:
```suggestion
The ``-example`` option creates an example application, and the CMake files needed to build it.
```

File: docs/fastddsgen/pubsub_app/includes/first_app.rst [hunk 3]
```diff
@@ -28,7 +18,7 @@ The `-example` option creates an example application, and the CMake files needed
 
             <path-to-Fast-DDS-workspace>/src/fastrtpsgen/scripts/fastrtpsgen -example CMake HelloWorld.idl
```
Comment (EduPonz) [inline docs/fastddsgen/pubsub_app/includes/first_app.rst:19]: I'm not sure, does `colcon` install this in the `install` dir?

File: docs/fastddsgen/pubsub_app/includes/first_app.rst [hunk 4]
```diff
@@ -42,13 +32,16 @@ The `-example` option creates an example application, and the CMake files needed
 
             <path-to-Fast-DDS-workspace>/src/fastrtpsgen/scripts/fastrtpsgen.bat -example CMake HelloWorld.idl
 
-        - However, if you have followed the **manual installation**, run:
+        - However, if you have followed the **stand-alone installation**, run:
 
         .. code:: bash
 
             <path-to-Fast-DDS-Gen>/scripts/fastrtpsgen.bat -example CMake HelloWorld.idl
```
Comment (EduPonz) [inline docs/fastddsgen/pubsub_app/includes/first_app.rst:39]: Where is this if binary installation? Maybe a note stating it?

File: docs/fastddsgen/pubsub_app/includes/first_app.rst [hunk 5]
```diff
@@ -42,13 +32,16 @@ The `-example` option creates an example application, and the CMake files needed
...
 
-Finally, we compile the generated code.
+Build the Fast DDS application
+********************************
+
+Then, compile the generated code.
```
Comment (EduPonz) [inline docs/fastddsgen/pubsub_app/includes/first_app.rst:44]: Where do I do this steps? The generated files, do I have to move them somewhere or are they generated in the running dir so I can build directly there? I'd include the creation and use of an example dir in these steps

File: docs/fastddsgen/pubsub_app/includes/first_app.rst [hunk 6]
```diff
@@ -93,9 +89,8 @@ You can also modify any values on your custom, IDL-generated data type.
    :language: c++
    :start-after: //PUBSUB_API_WRITE_SAMPLE
```
Comment (EduPonz) [inline docs/fastddsgen/pubsub_app/includes/first_app.rst:90]: Put a space after `//` in these snippet's comments

File: docs/fastddsgen/pubsub_app/includes/sum_and_next_steps.rst [hunk 7]
```diff
@@ -0,0 +1,12 @@
+Summary
```
Comment (EduPonz) [inline docs/fastddsgen/pubsub_app/includes/sum_and_next_steps.rst:1]: Combine these 2 into a single `summary and next steps` section

File: docs/fastddsgen/genuse/genuse.rst [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+
+Usage
+=====
+
+This section explains the usage of Fast DDS-Gen  tool and briefly describes the generated files.
+
+Running the Fast DDS-Gen Java application
+------------------------------------------
+
+First the steps outlined in :ref:`Linux installation of Fast DDS-Gen <fastddsgen_sl>` or
```
Comment (EduPonz) [inline docs/fastddsgen/genuse/genuse.rst:13]:
```suggestion
First, the steps outlined in :ref:`Linux installation of Fast DDS-Gen <fastddsgen_sl>` or
```

File: docs/fastddsgen/genuse/genuse.rst [hunk 9]
```diff
@@ -0,0 +1,75 @@
+.. include:: includes/aliases.rst
```
Comment (EduPonz) [inline docs/fastddsgen/genuse/genuse.rst:1]: I'd put this section before the example

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 10]
```diff
@@ -38,8 +53,8 @@ The following table shows the basic IDL types supported by *Fast DDS-Gen* and ho
 Arrays
 ^^^^^^
 
-*Fast DDS-Gen* supports unidimensional and multidimensional arrays.
-Arrays are always mapped to std::array containers.
+Fast DDS-Gen supports unidimensional and multidimensional arrays.
+Arrays are always mapped to ``std::array`` containers.
 The following table shows the array types supported and how they map.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:58]:
```suggestion
The following table shows the array types supported and their mapping.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 11]
```diff
@@ -115,7 +130,9 @@ You can define an IDL structure with a set of members with multiple types.
 It will be converted into a C++ class with each member mapped as an attribute plus methods to *get* and *set* each
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:130]: attribute/method -> data member/member function

File: code/FastDDSGenCodeTester.cpp [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+
+//STRUCTURES_DATA_TYPE
+class Structure
+{
+public:
+    Structure();
+    ~Structure();
+    Structure(const Structure &x);
+    Structure(Structure &&x);
+    Structure& operator=( const Structure &x);
```
Comment (EduPonz) [inline code/FastDDSGenCodeTester.cpp:15]:
```suggestion
    Structure& operator=(const Structure &x);
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 13]
```diff
@@ -251,34 +219,26 @@ The type MyBitset will store a total of 25 bits (3 + 10 + 12) and will require 3
 (lowest primitive type to store the bitset's size).
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:219]: :class:`MyBitset`

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 14]
```diff
@@ -251,34 +219,26 @@ The type MyBitset will store a total of 25 bits (3 + 10 + 12) and will require 3
...
+   :language: c++
+   :start-after: //BITSET_DATA_TYPE
+   :end-before: //!
 
-        void c(int32_t _c);
-        int32_t c() const;
-    private:
-        std::bitset<25> m_bitset;
-    };
+Internally is stored as a ``std::bitset``.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:232]:
```suggestion
Internally, it is stored as a ``std::bitset``.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 15]
```diff
@@ -290,17 +250,12 @@ Bitsets can inherit from other bitsets, extending their member set. ::
...
-    {
-        ...
-    };
+.. literalinclude:: /../code/FastDDSGenCodeTester.cpp
+   :language: c++
+   :start-after: //BITSET_INHERITANCE
+   :end-before: //!
 
 Note that in this case, ChildBitset will have two ``std::bitset`` members, one belonging to ParentBitset and the
 other belonging to ChildBitset.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:261]:
```suggestion
Note that in this case, :class:`ChildBitset` will have two ``std::bitset`` members, one belonging to :class:`ParentBitset` and the
other belonging to :class:`ChildBitset`.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 16]
```diff
@@ -311,7 +266,9 @@ Enumerations
 An enumeration in IDL format is a collection of identifiers that have a numeric value associated.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:266]:
```suggestion
An enumeration in IDL format is a collection of identifiers that have an associated numeric value.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 17]
```diff
@@ -347,28 +304,27 @@ The following IDL bitmask: ::
...
+.. literalinclude:: /../code/FastDDSGenCodeTester.cpp
+   :language: c++
+   :start-after: //BITMASK_DATA_TYPE
+   :end-before: //!
 
-The annotation *bit_bound* defines the width of the associated enumeration. It must be a positive number between
-1 and 64. If omitted, it will be 32 bits.
-For each *flag*, the user can use the annotation *position* to define the position of the flag. If omitted, it will
-be auto incremented from the last defined flag, starting at 0.
+The annotation *bit_bound* defines the width of the associated enumeration.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:314]:
```suggestion
The annotation ``bit_bound`` defines the width of the associated enumeration.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 18]
```diff
@@ -347,28 +304,27 @@ The following IDL bitmask: ::
...
+It must be a positive number between 1 and 64.
+If omitted, it will be 32 bits.
+For each *flag*, the user can use the annotation *position* to define the position of the flag.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:317]:
```suggestion
For each ``flag``, the user can use the annotation ``position`` to define the position of the flag.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 19]
```diff
@@ -347,28 +304,27 @@ The following IDL bitmask: ::
...
+If omitted, it will be auto incremented from the last defined flag, starting at 0.
 
 Keyed Types
 ^^^^^^^^^^^
 
 In order to use keyed topics, the user should define some key members inside the structure.
 This is achieved by writing “@Key” before the members of the structure you want to use as keys.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:324]:
```suggestion
This is achieved by writing the ``@Key`` annotation before the members of the structure you want to use as keys.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 20]
```diff
@@ -378,33 +334,34 @@ For example in the following IDL file the *id* and *type* field would be the key
...
 
 Including other IDL files
-^^^^^^^^^^^^^^^^^^^^^^^^^
-
-You can include another IDL files in yours in order to use data types defined in them. *Fast DDS-Gen* uses a C/C++
-preprocessor for this purpose, and you can use ``#include`` directive to include an IDL file.
+-------------------------
 
-.. code-block:: c
+You can include another IDL files in yours in order to use data types defined in them.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:344]:
```suggestion
You can include other IDL files in yours in order to use data types defined in them.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 21]
```diff
@@ -494,9 +451,12 @@ IDL 4.2 allows using the following names for primitive types:
 +------------------------+
 
 Forward declaration
-^^^^^^^^^^^^^^^^^^^
+---------------------
+
+Fast DDS-Gen supports forward declarations.
+As the example shows, this allows declaring inter-dependant structures, unions, etc.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:457]:
```suggestion
This allows declaring inter-dependant structures, unions, etc.
```

File: docs/fastddsgen/pubsub_app/includes/prerequisites.rst [hunk 22]
```diff
@@ -0,0 +1,8 @@
+Prerequisites
+^^^^^^^^^^^^^
+
+First of all, you need to follow the steps outlined in the Installation Manual for the installation of
```
[the remark this answers is not in the document]
Comment (rsanchez15) [reply docs/fastddsgen/pubsub_app/includes/prerequisites.rst:4]: After all, this is a tutorial so I would leave the references to the reader here.

File: docs/fastddsgen/pubsub_app/includes/first_app.rst (hunk 3 above)
Comment (rsanchez15) [reply docs/fastddsgen/pubsub_app/includes/first_app.rst:19]: Those scripts do not need installation and are downloaded when cloning the repositories. 
https://github.com/eProsima/Fast-DDS-Gen/tree/master/scripts

File: docs/fastddsgen/pubsub_app/includes/first_app.rst (hunk 4 above)
Comment (rsanchez15) [reply docs/fastddsgen/pubsub_app/includes/first_app.rst:39]: Check Windows installation directories.

File: docs/fastddsgen/pubsub_app/includes/first_app.rst (hunk 3 above)
Comment (EduPonz) [reply docs/fastddsgen/pubsub_app/includes/first_app.rst:19]: Sure, my question was more what happens when you `gradle assemble`. Do they get copied somewhere?

File: docs/fastddsgen/pubsub_app/includes/first_app.rst (hunk 4 above)
Comment (EduPonz) [reply docs/fastddsgen/pubsub_app/includes/first_app.rst:39]: Maybe we can point to somewhere where it say where `path-to-Fast-DDS-Gen` is if installing from binaries

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/514/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/515/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/516/)

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 23]
```diff
@@ -1,121 +1,139 @@
...
++-------------------------+--------------------------+
+| float a[5]              | std::array<float,5> a    |
++-------------------------+--------------------------+
+| double a[5]             | std::array<double,5> a   |
++-------------------------+--------------------------+
 
 Sequences
 ^^^^^^^^^
 
 *Fast DDS-Gen* supports sequences, which map into the STD vector container.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:87]: Just a suggestion
```suggestion
*Fast DDS-Gen* supports sequences, which map into the `std::vector` container.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 24]
```diff
@@ -1,121 +1,139 @@
...
++------------------------------+--------------------------+
+| sequence<float>              |    std::vector<float>    |
++------------------------------+--------------------------+
+| sequence<double>             |    std::vector<double>   |
++------------------------------+--------------------------+
 
 Maps
 ^^^^
 
 *Fast DDS-Gen* supports maps, which are equivalent to the STD map container.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:117]:
```suggestion
*Fast DDS-Gen* supports maps, which are equivalent to the `std::map` container.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 25]
```diff
@@ -347,28 +304,27 @@ The following IDL bitmask: ::
...
+For each ``flag``, the user can use the annotation ``position`` to define the position of the flag.
+If omitted, it will be auto incremented from the last defined flag, starting at 0.
 
 Keyed Types
 ^^^^^^^^^^^
 
 In order to use keyed topics, the user should define some key members inside the structure.
-This is achieved by writing “@Key” before the members of the structure you want to use as keys.
-For example in the following IDL file the *id* and *type* field would be the keys: ::
+This is achieved by writing the ``@Key`` annotation before the members of the structure you want to use as keys.
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:324]:
```suggestion
This is achieved by writing the ``@Key`` annotation before the members of the structure that are used as keys.
```

File: docs/fastddsgen/dataTypes/dataTypes.rst [hunk 26]
```diff
@@ -379,32 +335,34 @@ For example in the following IDL file the *id* and *type* field would be the key
...
 Including other IDL files
-^^^^^^^^^^^^^^^^^^^^^^^^^
-
-You can include another IDL files in yours in order to use data types defined in them. *Fast DDS-Gen* uses a C/C++
-preprocessor for this purpose, and you can use ``#include`` directive to include an IDL file.
+-------------------------
 
-.. code-block:: c
+Other IDL files can be included in addition to the current IDL file.
+*Fast DDS-Gen* uses a C/C++ preprocessor for this purpose, and you can use ``#include`` directive to include an IDL
```
Comment (EduPonz) [inline docs/fastddsgen/dataTypes/dataTypes.rst:345]:
```suggestion
*Fast DDS-Gen* uses a C/C++ preprocessor for this purpose, and ``#include`` directive can be used to include an IDL
```

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/517/)

**Comment by richiware** [comment]
Build status: [Build Status](http://jenkins.eprosima.com:8080/job/FastRTPS%20Docs%20Manual/518/)

**Pull request merged**